python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -124,13 +124,132 @@ def argsort_labels_values(labels_values, reverse: bool = False):
     :return: indices corresponding to the sorted values in ``labels_values``
     """
     if isinstance(labels_values, TorchTensor):
-        # torchscript does not support sorted for List[List[int]]
-        # so we temporary do this trick. this will be fixed with issue #366
-        max_int = torch.max(labels_values)
-        idx = torch.sum(
-            max_int ** torch.arange(labels_values.shape[1]) * labels_values, dim=1
-        )
-        return torch.argsort(idx, dim=-1, descending=reverse)
+        if labels_values.shape[1] == 1:
+            data = [(int(row[0]), i) for i, row in enumerate(labels_values)]
+            return torch.tensor(
+                [i[-1] for i in sort_list_2(data, reverse=reverse)],
```

Comment:
instead of adding `i` everywhere, can you do an argsort on the list of tuple?

Reply:
doesn't argsort exist only in numpy or similar?

Reply:
https://pytorch.org/docs/stable/generated/torch.argsort.html

Reply:
although this might only work with `Tensor`

Reply:
exactly sorting the tensor is not the same as sorting a list of tuples

Reply:
ok, all good then. Might be worth to add a comment there so I don't ask the question again next time I review this code =)

Reply:
Something like: 
"# Index is appended at the end to get the indices corresponding to the
  # sorted values. Because we append the indices at the end and since metadata
  # is unique, we do not affect the sorted order.
  # Using torch.argsort is not an option because only on Tensors and sorting a Tensor is 
  # different than sorting a List(Tuple)"
  
  ?

Reply:
that works!